mlir/test/Dialect/MIOpen/ops_f16.mlir

```diff
@@ -0,0 +1,196 @@
+// RUN: mlir-opt %s | FileCheck %s
+// RUN: mlir-opt %s | mlir-opt | FileCheck %s
+// Run: mlir-opt -mlir-print-op-generic %s | mlir-opt | FileCheck %s
+
```

Comment:
There isn't much fp16 dependent logic in those high level ops, consider creating 1 or 2 test case and attach at the corresponding places of the `ops.mlir`